### DISCUSSION TITLE
Custom 404 Page

### DISCUSSION THREAD

**Discussion by jiriteach** [NginxProxyManager/nginx-proxy-manager#1234] [Q&A]
Anyone know how to customise the 404 page? This is if I add a 404 host. 

Thanks

**Comment by jc21** [comment]
Currently this isn't possible using the app sorry. Will include it as a suggestion though

**Reply by jiriteach** [reply]
Thanks. I assume the change would have to be made inside the container or will be on the mapped folder? Can you point me to the location please? Thanks

**Reply by jc21** [reply]
This is the [404 nginx file template](https://github.com/jc21/nginx-proxy-manager/blob/develop/backend/templates/dead_host.conf).

For reference, here's the [default landing page template](https://github.com/jc21/nginx-proxy-manager/blob/develop/backend/templates/default.conf). This one will use a setting row within the template to decide what to show.

**Reply by jiriteach** [reply]
Awesome - Thanks

**Reply by ameinild** [reply]
I would challenge jc21, and say that it's partly possible. Here's what I did:

In my 404 host, under "Advanced", "Custom Nginx Configuration", I inserted this bit of code:

```
    error_page 404 /custom_404.html;
    location = /custom_404.html {
        root /data/nginx/custom;
        internal;
    }
```

Then, in the directory `/data/nginx/custom` (which may be mapped in your setup), I create the file `custom_404.html` with my custom 404 page.

This works perfectly for me. 👍

**Reply by jiriteach** [reply]
Tried this and its working well. Thanks for the pointer @ameinild.

**Reply by LIUYICHENG666** [reply]
> I would challenge jc21, and say that it's partly possible. Here's what I did:
> 
> In my 404 host, under "Advanced", "Custom Nginx Configuration", I inserted this bit of code:
> 
> ```
>     error_page 404 /custom_404.html;
>     location = /custom_404.html {
>         root /data/nginx/custom;
>         internal;
>     }
> ```
> 
> Then, in the directory `/data/nginx/custom` (which may be mapped in your setup), I create the file `custom_404.html` with my custom 404 page.
> 
> This works perfectly for me. 👍

Very good. The simplest solution.

**Comment by Guiorgy** [comment]
Every generated proxy configuration at the bottom has:

```text
# Custom
include /data/nginx/custom/server_proxy[.]conf;
```

So you could create the `data/nginx/custom/server_proxy.conf` (which should be mapped to a volume) and configure the 404 there.

This still doesn't work, however, when the hostname itself is not configured, in other words, you still get the default 404.
